File: A's/cpp/344A.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n; cin >> n;

    string previous;
    int groups = 0;
    while (n--) {
        string current; cin >> current;
        if (current != previous || previous.empty()) {
            groups++;
        }
        previous = current;
    }
    cout << groups << endl;
    
    return 0;
}

File: A's/py/344A.py
def main() -> None:
    n: int = int(input())

    previous: str = ""
    groups: int = 0
    while n:
        current: str = str(input())
        if current != previous or previous == "": groups += 1
        previous = current
        n -= 1
        
    print(groups)

if __name__ == "__main__":
    main()